### PR TITLE
HTML Reporter: Make it more obvious why diff is suppressed

### PR DESCRIPTION
**Pull request by shivamdixit**
If the depth of an object is more than 'Max Depth' specified in `Qunit.dump` then the diff is not displayed. This behavior is confusing. This PR will display a message to the user so that user knows why it is not displayed.

Refer: https://github.com/jquery/qunit/issues/793#issuecomment-84495125

I'm confused with what text to display to the user since the user might not be aware of max depth. Any suggestions?

Screenshot:
qunit4

### CODE REVIEW

**Comment by scottgonzalez** [comment]
Ideally it says what the current max depth is and provides a link to rerun with a higher max depth or no max depth.

**Comment by shivamdixit** [comment]
@scottgonzalez I've updated the PR. Screenshot:

qunit5

1. Added `maxDepth` in url parameter, `maxDepth = -1` implies run the tests without any `maxDepth`.

2. Although `maxDepth = someVal` in URL parameters will work, but not *explicitly* showing the option of inputting the value from user because the UI will be cluttered as we will need to add input box, button etc.

3. Instead showing `QUnit.dump.maxDepth` as hint, in case user wants to change `maxDepth`.

4. The color of `Rerun` looks dull in the message, any suggestions for a different color?

5. As `QUnit.dump` is not in scope when we are parsing URL params, hence I've added extra property `QUnit.config.maxDepth`.

@jzaefferer Please provide your valuable suggestions as well.

**Comment by shivamdixit** [comment]
I think the tests are failing because of the timeout. How can I re-run the tests?

File: reporter/html.js [hunk 1]
```diff
@@ -663,6 +663,15 @@ QUnit.log(function( details ) {
 				actual + "</pre></td></tr>" +
 				"<tr class='test-diff'><th>Diff: </th><td><pre>" +
 				QUnit.diff( expected, actual ) + "</pre></td></tr>";
+		} else {
+			if ( expected.indexOf( "[object Array]" ) !== -1 ||
+					expected.indexOf( "[object Object]" ) !== -1 ) {
+				message += "<tr class='test-message'><th>Message: </th><td>" +
+					"Diff suppressed as the depth of object is more than current max depth (" +
+					QUnit.config.maxDepth + ").<p>Hint: Use 'QUnit.dump.maxDepth' to run " +
+					" with a higher max depth or <a href='?maxDepth=-1'>" +
```
Comment (jzaefferer) [inline reporter/html.js]: This should use `setUrl()` to generate QUnit specific URLs that include other URL parameters already present (filters etc.).

**Comment by jzaefferer** [comment]
@shivamdixit thanks for the PR. I have very little bandwidth to review anything this week. I added one comment above. I'd like to discuss the general concept for tackling this issue, but that has to wait a few days.

**Comment by scottgonzalez** [comment]
That screenshot looks great. You should wrap `QUnit.config.maxDepth` in `<code>` instead of quotes.

Fixing the styling of the rerun link should be very simple, but let's wait for the team to discuss.

**Comment by shivamdixit** [comment]
@jzaefferer @scottgonzalez Thanks for your suggestions. I've updated the code.

**Comment by jzaefferer** [comment]
Let's add some CSS to display the Rerun link properly, just change `color` to something with more contrast.

**Comment by shivamdixit** [comment]
qunit5

How is it looking?

**Comment by leobalter** [comment]
LGTM

**Pull request closed without merging**